Makefile

```diff
@@ -12,7 +12,17 @@ export GOPATH = $(GODEPS):$(USER_GOPATH)
 
 DEV_DEPS = github.com/uber/tchannel-go/thrift/thrift-gen \
 		   github.com/vektra/mockery/... \
-		   github.com/tools/godep
+		   github.com/tools/godep \
+		   github.com/golang/lint/golint
+
+# Automatically gather packages
+PKGS = $(shell find . -type d -maxdepth 3 \
+	! -path '*/.git*' \
+	! -path '*/_*' \
+	! -path '*/Godeps*' \
+	! -path '*/test*' \
+	! -path '*/examples*' \
+)
```

Comment:
`/test/` yes - it includes mocks that have lint errors that are outside our control. Since they're only test utilities they're less important.

`/examples` - this one is harder to justify excluding. I checked again and actually the lint errors from the examples are mostly in generated code. So I've updated the PR to exclude the generated code (instead of the entire examples directory).